builder/js/ncf.js

```diff
@@ -431,10 +436,74 @@ $scope.groupMethodsByCategory = function () {
       // Select the technique, by using angular.copy to have different objects
       $scope.selectedTechnique=angular.copy(technique);
       $scope.originalTechnique=angular.copy($scope.selectedTechnique);
+
       $scope.$broadcast('endSaving');
     }
   };
 
+  /* --- LOCAL STORAGE --- */
+
+  /*
+    Here are the different cases to cover :
+      - Modify a technique / Create a new technique, then reload the page without saving it.
+      - Start editing a technique, modify it and save it on another navigator, then reload the page without saving it :
+        - Save it with your changes.
+        - Save it with after update it.
+        - Save it with the same name than the modified version.
+        - Update the technique with the latest changes.
+  */
+  $scope.updateItemSessionStorage = function(item, oldTechnique, newTechnique){
+    if(oldTechnique && !newTechnique){
```

Comment:
Please do check on "undefined" not if they are truthy, this leads to javascript major failures !!

> oldtechnique !== undefined && newTechnique === undefined